app/_service/download.service.ts

```diff
@@ -55,23 +42,25 @@ export class DownloadService {
 
     for (let key of unique){
       let headers = headers4type[key];
-      csvFile += addTitle(key, headers);
+      csvFile += Papa.unparse(Array(headers));
```

Comment:
Have you tried using the PapaParse.unparse method variant that gets the list of fields and the data as an array of objects?
Might be easier to use that this loop.

Reply:
Either way I think that rewriting this using the `map` method would make the code much more readable.